### PR TITLE
Adding support for local, custom ADCIRC src.

### LINKED ISSUE #748: support adding untracked ADCIRC patchsets
Also generate menu options dynamically based on contents of the directories contained in `$ADCIRC_PATCH_PATH`.

### PR DESCRIPTION
**Pull request by wwlwpd**
Issue-748: This will allow users with ADCIRC source somewhere on their
ASGS host machine to build and register ADCIRC via `build adcirc`, by
selecting the "custom" option and answering the questions in the interactive
mode that follows. Upon successful compilation, this ADCIRC is available
like all others via "load adcirc" and can be seen along with the others
via "list adcirc". Patches are not supported at this time.

Resolves #748.

### CODE REVIEW

**Comment by wwlwpd** [comment]
@jasonfleming this needs to be tested _and_reviewed. @notstarboard should also take a look at it. Let me know when y'all can discuss and I can describe the testing procedure; really just requires local file access to the original "source" (the original source is left untouched, just `cp`'d).

**Comment by jasonfleming** [comment]
Will work on this on hatteras; need a copy of the source code ; copy to work dir where asgs is setup ; start shell ; run build adcirc ; last menu option is custom ; (11 for now) ; will ask for path to source ; tld  not work ; ask for name etc ; skips patching and git checkout ; copies source from you dir to normal asgs place ; from there identical to previous procedure ; adds to adcirc meta dir file with info needed ; also produces json file ; new key called `custom_src` plus env var ; if it is pointing to custom source this has full path ; otherwise 0 value ; once built successfully works just like any other adcirc build ; `load adcirc x` etc ; will get path to source via email

**Comment by wwlwpd** [comment]
rebased on current master SHA, f3f89d82687802d88f79a301213abc73c19aa9b2

**Comment by jasonfleming** [comment]
Thank you @wwlwpd I will be testing this today and providing feedback.

**Comment by wwlwpd** [comment]
@dave-gso has reported an interest issue, I will try to reproduce and fix if found.

File: bin/init-adcirc.sh [hunk 1]
```diff
@@ -190,38 +239,56 @@ ADCSWAN_MAKE_CMD="make $ADCSWAN_BINS compiler=${ADCIRC_COMPILER} MACHINENAME=${A
...
+# git repo wrangling if not copying from a local filesystem
+#TODO: put all this in function
+if [[ -d "${CUSTOM_SRC}" ]]; then                     # handle custom filesystem based source, no patches
+  # copy across local file system
+  mkdir -p $ADCIRCBASE
+  echo " ... copying from from '$CUSTOM_SRC' to '$ADCIRCBASE'"
+  cp -rf $CUSTOM_SRC/* $ADCIRCBASE
+  # clean up destination
+  echo " ... cleaning up cruft from '$ADCIRCBASE'"
+  rm -rf $ADCIRCBASE/.git                2> /dev/null
```
Comment (jasonfleming) [inline bin/init-adcirc.sh:251]: suggest to use `make clean` or similar

**Review by jasonfleming — DISMISSED** [review]
This looks great @wwlwpd thank you. Suggest additional comments to document any assumptions or caveats, e.g., that the `make` build method is used (not cmake), and that the netcdf paths should not be hard coded in the developers' build system.

File: bin/init-adcirc.sh (hunk 1 above)
Comment (wwlwpd) [reply bin/init-adcirc.sh:251]: good idea, will do!

**Comment by wwlwpd** [comment]
@jasonfleming I will commit to completing #975, to document `build adcirc` in the wiki once this is merged - just to keep the scope of the work down.

**Comment by wwlwpd** [comment]
Latest commit,

```
commit 708c8328a3bff0573e5c125b0359246f1264074c
Author: wwlwpd <46434714+wwlwpd@users.noreply.github.com>
Date:   Fri Aug 12 04:09:28 2022 -0400

    fixup! Adding support for custom paths to ADCIRC src.
    
    Replaced series of `rm -rv` with `make clobber`; detects if
    target directory exists, and if so will offer to delete it. The
    default for this is set to N (no). Added fun Star Wars ASCII art
    that is shown whenever ADCIRC build succeeds.
```

**Comment by wwlwpd** [comment]
Preliminary documentatio, https://github.com/StormSurgeLive/asgs/wiki/ASGS-Compiling-ADCIRC

**Review by jasonfleming — APPROVED** [review]
Hey @wwlwpd I tested this on hatteras, and was able to build @dave-gso weir_pump code successfully. I executed `adcirc` on the command line in the asgs shell and it was able to start successfully without any library errors. I was then able to build v55.01 (bonus testing) successfully and test run `adcirc` on the command line. Looks great!

**Comment by wwlwpd** [comment]
Thanks for testing, will squash and merge!

**Comment by wwlwpd** [comment]
Merged manually. Approved by @jasonfleming

**Pull request closed without merging**